auslib/web/common/releases.py

```diff
@@ -51,16 +52,23 @@ def _get_release(release):
     return releases[0] if releases else None
 
 
-def get_release(release):
+def get_release(release, with_csrf_header=False):
     release = _get_release(release)
     if not release:
-        return Response(status=404, mimetype="application/json")
+        return problem(404, "Not Found", "Release name: %s not found" % release)
     headers = {'X-Data-Version': release['data_version']}
+    if with_csrf_header:
+        headers.update(get_csrf_headers())
     if request.args.get("pretty"):
         indent = 4
     else:
         indent = None
-    return Response(response=json.dumps(release['data'], indent=indent, sort_keys=True), mimetype='application/json', headers=headers)
+    return Response(response=json.dumps(release['data'], indent=indent, sort_keys=True),
+                    mimetype='application/json', headers=headers)
+
+
+def get_release_with_csrf_header(release):
+    return get_release(release, with_csrf_header=True)
```

Comment:
I think these would be better written like:
`get_release_with_csrf_header = functools.partial(get_release, with_csrf_header=True)`

That way, this function won't need to change if the args to underlying functions change, and it removes any temptation to modify the response returned.

Reply:
I need keep this as is, because inspect.getargspec function does not supports partial object:
`    auslib/test/admin/views/test_rules.py:5: in <module>
        from auslib.test.admin.views.base import ViewTest
    auslib/test/admin/views/base.py:8: in <module>
        from auslib.web.admin.base import app
    auslib/web/admin/base.py:29: in <module>
        connexion_app.add_api(spec, validate_responses=True, strict_validation=True)
    .tox/py27/local/lib/python2.7/site-packages/connexion/apps/flask_app.py:54: in add_api
        api = super(FlaskApp, self).add_api(specification, **kwargs)
    .tox/py27/local/lib/python2.7/site-packages/connexion/apps/abstract.py:159: in add_api
        options=api_options.as_dict())
    .tox/py27/local/lib/python2.7/site-packages/connexion/apis/abstract.py:131: in __init__
        self.add_paths()
    .tox/py27/local/lib/python2.7/site-packages/connexion/apis/abstract.py:252: in add_paths
        self._handle_add_operation_error(path, method, sys.exc_info())
    .tox/py27/local/lib/python2.7/site-packages/connexion/apis/abstract.py:263: in _handle_add_operation_error
        six.reraise(*exc_info)
    .tox/py27/local/lib/python2.7/site-packages/connexion/apis/abstract.py:242: in add_paths
        self.add_operation(method, path, endpoint, path_parameters)
    .tox/py27/local/lib/python2.7/site-packages/connexion/apis/abstract.py:196: in add_operation
        self._add_operation_internal(method, path, operation)
    .tox/py27/local/lib/python2.7/site-packages/connexion/apis/flask_api.py:87: in _add_operation_internal
        function = operation.function
    .tox/py27/local/lib/python2.7/site-packages/connexion/operation.py:368: in function
        self.parameters, self.consumes, self.__undecorated_function, self.pythonic_params)
    .tox/py27/local/lib/python2.7/site-packages/connexion/decorators/parameter.py:111: in parameter_to_arg
        arguments, has_kwargs = inspect_function_arguments(function)
    .tox/py27/local/lib/python2.7/site-packages/connexion/decorators/parameter.py:45: in inspect_function_arguments
        argspec = inspect.getargspec(function)
    /usr/lib/python2.7/inspect.py:816: in getargspec
        raise TypeError('{!r} is not a Python function'.format(func))
    E   TypeError: <functools.partial object at 0x7f05d3c79c58> is not a Python function `

Reply:
Ah, that's too bad. No big though.